packages/babel-parser/src/parser/statement.js

```diff
@@ -1349,7 +1349,8 @@ export default class StatementParser extends ExpressionParser {
     // Check the key is not a computed expression or string literal.
     const isSimple = key.type === "Identifier";
 
-    this.parsePostMemberNameModifiers(publicMember);
+    // "async" should not be treated as normal porperty names
+    if (key.name !== "async") this.parsePostMemberNameModifiers(publicMember);
```

Comment:
`async` could be a class property, i.e.
```ts
class A {
  async?: boolean
}
```

Reply:
Wow nice, thank you

Reply:
hey @JLHwung i was trying to follow this suggestion, but i'm not sure of the proper way to check if a token was eaten, this is what i've tried:

```javascript
let token;
if ((token = this.state.tokens.find(s => s.type === tt.question))) {
  this.unexpected(token.start);
}
```

with this modification to `parsePostMemberNameModifiers`

```javascript
parsePostMemberNameModifiers(
  methodOrProp: N.ClassMethod | N.ClassProperty,
  saveToken: boolean,
): void {
  this.options.tokens = saveToken;
  const optional = this.eat(tt.question);
  if (optional) methodOrProp.optional = true;
} 
```
the problem with that is that previous tests are failing because the absence of the property `options.tokens` in their `output.json` files

thanks you! 🙏

Reply:
```js
if ((token = this.state.tokens.find(s => s.type === tt.question))) {
  this.unexpected(token.start);
}
```
I am afraid we may not iterate `this.state.tokens` here because it may run out of the public member method scope, i.e. when parsing `foo()` it would throw on the previous consumed `?` token, which is incorrect.
```
class A {
  async?: boolean,
  async foo(): Promise<void>
}
```

You may consider to log the `maybeQuestionTokenStart = this.state.start` before the first `parsePostMemberNameModifiers` call. After that, if we are in an `async` method later, we may check if `publicMethod.optional` is true. If it is, then it must come from the `?` token after `async` keyword, and now we should throw at `maybeQuestionTokenStart`.

Reply:
This is awesome, thanks a lot for taking the time and the explanation, i've updated the PR